Revert gold necklace name in beecoin shop

Sinmero opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
After the name of the gold necklace was changed those who bought it before lost access to it (at least i did).
Reverts beecoin shop gold necklace name from 'dope necklace' to 'necklace, gold'.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Those who purchased gold necklace before the name change will have access to it again (hopefully).

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: People who bought gold necklace in beecoin shop before its name change should have access to it again.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sinmero committed:
Revert gold necklace beecoing shop name

PowerfulBacon reviewed (dismissed):
I like the change, since dope necklace is kind of a stupid name, but the shop really should not be based off of display name if it is.

Sinmero commented:
This is how i believe it works: client recieves data with the list of beecoin items your account has from the server and since the old necklace name doesnt match with the new one it doesnt display that you own one.

ike709 commented:
It does apparently use display names. However that means that anybody that bought it after it got renamed will lose it.

Sinmero committed:
fix

beecoin shop now has two names for gold necklace: 'dope necklace' and 'necklace, gold'

Sinmero commented:
I added both variants to beecoin shop as a temporary solution. Someone with sever access should fix it properly thou.
